numpy/core/src/common/npy_cpu_features.c

```diff
@@ -551,7 +563,26 @@ npy__cpu_init_features(void)
     npy__cpu_have[NPY_CPU_FEATURE_VSX2] = (hwcap & PPC_FEATURE2_ARCH_2_07) != 0;
     npy__cpu_have[NPY_CPU_FEATURE_VSX3] = (hwcap & PPC_FEATURE2_ARCH_3_00) != 0;
     npy__cpu_have[NPY_CPU_FEATURE_VSX4] = (hwcap & PPC_FEATURE2_ARCH_3_1) != 0;
-// TODO: AIX, FreeBSD
+#elif defined(__FreeBSD__)
+    unsigned long hwcap;
+    elf_aux_info(AT_HWCAP, &hwcap, sizeof(hwcap));
+    if ((hwcap & PPC_FEATURE_HAS_VSX) == 0)
+        return;
+
+    elf_aux_info(AT_HWCAP2, &hwcap, sizeof(hwcap));
```

Comment:
```suggestion
#if defined(__linux__) || defined(__FreeBSD__)
#ifdef __linux__
    unsigned int hwcap = getauxval(AT_HWCAP);
    if ((hwcap & PPC_FEATURE_HAS_VSX) == 0)
        return;

    hwcap = getauxval(AT_HWCAP2);
#else
    unsigned long hwcap;
    elf_aux_info(AT_HWCAP, &hwcap, sizeof(hwcap));
    if ((hwcap & PPC_FEATURE_HAS_VSX) == 0)
        return;

    elf_aux_info(AT_HWCAP2, &hwcap, sizeof(hwcap));
#endif // __linux__
```
To reuse the following lines for both platforms.

Reply:
That fails to build with:
```
numpy/core/src/common/npy_cpu_features.c:545:18: error: use of undeclared identifier 'PPC_FEATURE_HAS_VSX'
    if ((hwcap & PPC_FEATURE_HAS_VSX) == 0)
                 ^
1 error generated.

`PPC_FEATURE_HAS_VSX` is defined in `machine/cpu.h`.
```